core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -197,16 +190,24 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
         _clusterId = getOrGenerateClusterId(zkUtils)
         info(s"Cluster ID = $clusterId")
 
+        /* generate brokerId */
+        config.brokerId =  getBrokerId
+        this.logIdent = "[Kafka Server " + config.brokerId + "], "
+
+        /* create and configure metrics */
+        val reporters = config.getConfiguredInstances(KafkaConfig.MetricReporterClassesProp, classOf[MetricsReporter],
+            Map[String, AnyRef](KafkaConfig.BrokerIdProp -> config.brokerId.asInstanceOf[AnyRef]).asJava)
+        reporters.add(new JmxReporter(jmxPrefix))
```

Comment:
We call Metrics.close() during shutdown, which already calls close() on all metric reporters.